### PR TITLE
chore: `bundle install` with `--with` option is dispensable in this case

### PR DESCRIPTION
**Pull request by sato11**
TLDR; `bundle install` and `bundle install --with development` both install the same set. So perhaps one without flag is clearer to use?

The setup script and the ci workflow definition specify that bundler should use `--with development` option, which has been deprecated as of newer version of bundler. Mine is 2.4.3.

```
[DEPRECATED] The `--with` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set --local with 'development'`, and stop using this flag
```

While the intention is apparently to restrict installed gems to development group, it makes no difference to the resulting installation set. 

This is because we cannot rely on `--with` to opt out from installing grouped gems. What we might want to use is either `--without` option or `group :test, optional: true`. A relevant reference is here:
https://bundler.io/guides/groups.html#optional-groups-and-bundlewith

Here I propose that we drop the irrelevant flag. The discussion above in turn shows that  `gem 'simplecov'` does not need to reside in test group, as it's been, and will be, installed regardless of the presence of `--with`. Therefore it's got rid of as well.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-ruby/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

**Comment by shrutiburman** [comment]
Thanks for the PR! We've added the review to our backlog to be prioritised. Pull requests and +1s on the issue summary will help it move up the backlog.
